modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/runtime/streaming/PagingProviderProducer.java

```diff
@@ -94,13 +98,13 @@ public int getSize() {
    * Closes the delegate
    */
   @Override
-  public void close() throws IOException {
+  public void close() {
     ConnectionSupplier connectionSupplier = null;
     try {
       connectionSupplier = connectionSupplierFactory.getConnectionSupplier();
       delegate.close(connectionSupplier.getConnection());
     } catch (Exception e) {
-      throw new MuleRuntimeException(createStaticMessage(COULD_NOT_OBTAIN_A_CONNECTION), e);
+      throw new MuleRuntimeException(createStaticMessage(PAGE_ERROR), e);
```

Comment:
this is loosing information. You should still thrwo the COULD_NOT_OBTAIN_A_CONNECTION if `connectionSupplier.getConnection()` fails. Agree to throw PAGE_ERROR in the other cases